OpenTracing/OpenTracing.OpenTracing/ISpan.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace OpenTracing
+{
+    public interface ISpan<T> 
+    {
+        T GetSpanContext();
```

Comment:
This (and the templating on a `T` context type) is the most controversial thing in this PR, as @yurishkuro pointed out. Without getting into a discussion (yet) of whether it's better or worse, are you able to quickly describe how things would be different if you got rid of the `<T>` and the `GetSpanContext()`?